crates/ide/src/hover/render.rs

```diff
@@ -964,3 +980,75 @@ fn keyword_hints(
         _ => KeywordHint::new(token.text().to_owned(), format!("{}_keyword", token.text())),
     }
 }
+
+fn render_object_safety(
+    db: &RootDatabase,
+    name: Name,
+    safety: Option<ObjectSafetyViolation>,
+) -> SmallVec<[String; 3]> {
```

Comment:
```suggestion
fn render_object_safety(
    db: &RootDatabase,
    buf: &mut String,
    name: Name,
    safety: Option<ObjectSafetyViolation>,
) {
```
and then just use `format_to!(buf, ...)` to write into that. Should be a bit easier to read